dev/v1-to-v2/v1_to_v2.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+## Copyright 2025, gRPC Authors All rights reserved.
+##
+## Licensed under the Apache License, Version 2.0 (the "License");
+## you may not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+##     http://www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+
+set -eou pipefail
+
+log() { printf -- "** %s\n" "$*" >&2; }
+error() { printf -- "** ERROR: %s\n" "$*" >&2; }
+fatal() { error "$@"; exit 1; }
+
+# Clones v1 into the given directory and applies a number of patches to rename
+# the package from 'grpc-swift' to 'grpc-swift-v1' and 'protoc-gen-grpc-swift'
+# to 'protoc-gen-grpc-swift-v1'.
+function checkout_v1 {
+  # The directory to clone grpc-swift into.
+  grpc_checkout_dir="$(realpath "$1")"
+  # The path of the checkout.
+  grpc_checkout_path="${grpc_checkout_dir}/grpc-swift-v1"
+  # Version of grpc-swift to checkout.
+  grpc_v1_tag="1.24.2"
```

Comment:
Would it be worth getting the latest tag by curling GH instead of hardcoding this here?

Reply:
Yeah, we can just look at the tags though.